### PR TITLE
Ch api test setup 167313391

### PR DESCRIPTION
**Pull request by Anguandia**
#### What does this PR do?
  
  Setup unit testing for the backend

#### Description of Task to be completed?

  - Add required dependancies for running tests
  - Create test directory with general test setup module
  - Update test running script
  - Test the index route

#### How should this be manually tested?

  - Clone this repo and run git checkout -b ch-api-test-setup-167313391
  - Run git pull ch-api-test-setup-167313391
  - Run npm test

#### Any background context you want to provide?

  Team members working on different features should be able to build their tests on a common platform as:
  - Open/Create a file for the given tests in the test directory, es test users
  - Import {chai, app} from ./test-setuo
  - Ensure that required modules/functions not in the test-setup package are imported from their respective sources
  - Write the test suite(s)/cases

#### What are the relevant pivotal tracker stories?

  [finishes #167313391](https://www.pivotaltracker.com/story/show/167313391)

### CODE REVIEW

**Review by Mnickii — APPROVED** [review]
Lgtm

**Review by nkalyesubula — APPROVED** [review]

**Review by nshutijonathan — APPROVED** [review]
for me It looks good

**Review by habinezadalvan — COMMENTED** [review]
@Anguandia I saw that you used **Should** assertion and I was trying to read about the difference between **expect** and **should** assertions and I found that they seem to work in the same way; however, expect is browser compatible while should is not. What do you think if you use **expect** instead of **should**?

**Review by minega25 — APPROVED** [review]

File: test/test-setuo.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+
+import server from '../src/app';
+
+
+chai.use(chaiHttp);
+should = chai.should();
```
Comment (Mnickii) [inline test/test-setuo.js]: Let's use expect for our assertions.

**Review by Mnickii — CHANGES_REQUESTED** [review]

File: test/test-setuo.js (hunk 1 above)
Comment (Anguandia) [reply test/test-setuo.js]: Done, now using expect

File: README.md [hunk 2]
```diff
@@ -1,6 +1,11 @@
 [![Build Status](https://travis-ci.org/andela/codepirates-ah-backend.svg?branch=develop)](https://travis-ci.org/andela/codepirates-ah-backend)
+[![Coverage Status](https://coveralls.io/repos/github/andela/codepirates-ah-backend/badge.svg?branch=develop)](https://coveralls.io/github/andela/codepirates-ah-backend?branch=develop)
 
 # Authors Haven - A Social platform for the creative at heart.
+[![Reviewed by Hound CI](https://img.shields.io/badge/Reviewed%20by-Hound%20CI-blue.svg)](https://houndci.com)
+
+Authors Haven - A Social platform for the creative at heart.
```
Comment (Mnickii) [inline README.md]: I believe this last title was to be removed

File: .hound.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+eslint:
+  enabled: true
+  config_file: .eslintrc
+
```
Comment (Mnickii) [inline .hound.yml:4]: This work was already merged into develop in another pr. Please remove it

**Review by Mnickii — CHANGES_REQUESTED** [review]

**Pull request merged**